File: hptuScoreClient/src/app/dto/QuestionSummaryDto.ts
import { FormArray } from "@angular/forms";
import { QuestionDto } from "./QuestionDto";
import { ScoreSummarySetUpDto } from "./ScoreSummarySetUpDto";
import { HptuAssessmentDto } from "./HptuAssessmentDto";

export interface QuestionSummaryDto {
    id: number;
    summary: string;
    minimumPreviousQuestionScore: number;
    questionOrderNumber: number;
    functionalityId: number;
    questions: QuestionDto[]
    scoreSummaries: ScoreSummarySetUpDto[]
    dataSource: HptuAssessmentDto[]
    formArray: FormArray;
}

File: hptu-api/src/main/java/com/hptu/functionality/dto/QuestionSummaryDto.java
package com.hptu.functionality.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.io.Serializable;
import java.util.List;

public record QuestionSummaryDto(
        Long id,
        @NotBlank String summary,
        @NotNull Integer minimumPreviousQuestionScore,
        @NotNull Integer questionOrderNumber,
        Long functionalityId,
        List<QuestionDto> questions,
        List<ScoreSummarySetUpDto> scoreSummaries
) implements Serializable, Comparable<QuestionSummaryDto> {
        @Override
        public int compareTo(QuestionSummaryDto o) {
                return Integer.compare(this.questionOrderNumber, o.questionOrderNumber());
        }
}
